### PR TITLE
feat(community): consolidate community funds

### PR DESCRIPTION
**Pull request by drklee3**
## Description
- Transfer `x/distribution` communitypool and `x/kavadist` (non-ukava) account funds to `x/community`

## Checklist
 - [x] Changelog has been updated as necessary.

### CODE REVIEW

File: x/community/keeper/incentives.go [hunk 1]
```diff
@@ -51,7 +58,85 @@ func (k Keeper) StartDisableInflationUpgrade(ctx sdk.Context) {
...
+	}
+
+	logger.Info(fmt.Sprintf("transferred %s from x/distribution to x/community", truncatedCoins))
+
+	// Set x/distribution community pool to remaining dust amounts
+	k.distrKeeper.SetFeePool(ctx, distrtypes.FeePool{
+		CommunityPool: leftoverDust,
+	})
+
+	logger.Info("updated x/distribution community pool to 0")
```
Comment (DracoLi) [inline x/community/keeper/incentives.go:115]: this shouldn't be exact 0 because of dust. Perhaps print out the dust amount here?

**Review by DracoLi — APPROVED** [review]
looks great

File: x/community/keeper/incentives_test.go [hunk 2]
```diff
@@ -134,9 +139,147 @@ func (suite *IncentivesTestSuite) TestStartDisableInflationUpgrade() {
...
+
 func (suite *IncentivesTestSuite) setUpgradeTimeFromNow(t time.Duration) {
 	params, found := suite.Keeper.GetParams(suite.Ctx)
 	suite.True(found)
 	params.UpgradeTimeDisableInflation = suite.Ctx.BlockTime().Add(t)
 	suite.Keeper.SetParams(suite.Ctx, params)
 }
+
+// EventsContains asserts that the expected event is in the provided events
+func (suite *IncentivesTestSuite) EventsContains(events sdk.Events, expectedEvent sdk.Event) {
```
Comment (pirtleshell) [inline x/community/keeper/incentives_test.go:264]: could refactor or remove the suite dependency and add this to the common test helpers for use across the whole repo?

File: x/community/keeper/incentives_test.go [hunk 3]
```diff
@@ -134,9 +139,147 @@ func (suite *IncentivesTestSuite) TestStartDisableInflationUpgrade() {
...
+func (suite *IncentivesTestSuite) TestStartCommunityFundConsolidation() {
+	suite.SetupTest()
+	ak := suite.App.GetAccountKeeper()
+
+	initialFeePool := distrtypes.FeePool{
+		CommunityPool: sdk.NewDecCoins(
+			sdk.NewDecCoinFromDec("ukava", sdk.NewDecWithPrec(123456, 2)),
+			sdk.NewDecCoinFromDec("usdx", sdk.NewDecWithPrec(654321, 3)),
+		),
+	}
```
Comment (pirtleshell) [inline x/community/keeper/incentives_test.go:151]: i think we should add an additional test case where the fee pool is `distrtypes.InitialFeePool()` (zero) as a sanity check this works on fresh chains that will have an empty fee pool.

File: x/community/testutil/consolidate.go [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+
+	distrtypes "github.com/cosmos/cosmos-sdk/x/distribution/types"
+	"github.com/kava-labs/kava/app"
+	types "github.com/kava-labs/kava/x/community/types"
+	kavadisttypes "github.com/kava-labs/kava/x/kavadist/types"
+)
+
+func (suite *disableInflationTestSuite) TestStartCommunityFundConsolidation() {
```
Comment (nddeluca) [inline x/community/testutil/consolidate.go:15]: great coverage 👍

File: x/community/migrations/v2/store.go [hunk 5]
```diff
@@ -11,10 +11,6 @@ import (
 	"github.com/kava-labs/kava/x/community/types"
 )
 
-const (
-	ModuleName = "mint"
```
Comment (nddeluca) [inline x/community/migrations/v2/store.go:15]: good catch

File: x/community/testutil/consolidate.go [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+				fundCoins,
+			)
+			suite.NoError(err, "x/distribution account should be funded without error")
+
+			err = suite.App.FundModuleAccount(
+				suite.Ctx,
+				kavadisttypes.ModuleName,
+				tc.initialKavadistBalance,
+			)
+			suite.NoError(err, "x/kavadist account should be funded without error")
```
Comment (nddeluca) [inline x/community/testutil/consolidate.go:85]: nice error messages, very helpful

File: x/incentive/keeper/msg_server_hard_test.go [hunk 7]
```diff
@@ -40,7 +40,7 @@ func (suite *HandlerTestSuite) TestPayoutHardClaimMultiDenom() {
 
 	// Claim denoms
 	err := suite.DeliverIncentiveMsg(&msg)
-	suite.NoError(err)
+	suite.Require().NoError(err)
```
Comment (nddeluca) [inline x/incentive/keeper/msg_server_hard_test.go:43]: 👍

**Review by nddeluca — APPROVED** [review]

**Pull request merged**